blob/s3blob/s3blob.go

```diff
@@ -225,12 +238,20 @@ func (b *bucket) NewTypedWriter(ctx context.Context, key string, contentType str
 			u.PartSize = int64(opts.BufferSize)
 		}
 	})
+	var metadata map[string]*string
+	if opts != nil && len(opts.Metadata) > 0 {
+		metadata = make(map[string]*string, len(opts.Metadata))
+		for k, v := range opts.Metadata {
+			metadata[k] = aws.String(v)
```

Comment:
Same here: this should return an error if the key is like `Content-Type`.

Reply:
See above.